editor/src/components/canvas/dom-lookup.ts

```diff
@@ -61,25 +89,43 @@ export function findFirstParentWithValidElementPath(
 
   const filteredValidPathsMappedToDynamic = mapDropNulls(
     (validPath: string) => {
-      return staticAndDynamicTargetElementPaths.find(
-        (staticAndDynamic) => staticAndDynamic.static === validPath,
-      )?.dynamic
+      if (allowDescendantsFromPath) {
+        for (const staticAndDynamic of staticAndDynamicTargetElementPaths) {
+          if (EP.isDescendantOfOrEqualTo(staticAndDynamic.staticPath, EP.fromString(validPath))) {
+            const depthDiff =
+              EP.navigatorDepth(staticAndDynamic.staticPath) -
+              EP.navigatorDepth(EP.fromString(validPath))
+            let ancestor = staticAndDynamic.dynamic
+            for (let i = 0; i < depthDiff; i++) {
+              ancestor = EP.parentPath(ancestor)
+            }
```

Comment:
@gbalint I think it's worth turning this into a helper function in element-path.ts, and figuring out what was wrong with the function that's already there and seemed totally broken

Reply:
I extracted a function here: d611beeb58efe534fc680632df3b733bb35a4042
I believe the original `getNthParent` function is not broken, just does something completely different: it goes n levels up in the component hierarchy.
I did not wanna touch that because maybe that is what is needed there. However, I still believe we should make our terminology clearer, let's discuss that offline.